backend/db/utils.py

```diff
@@ -35,27 +36,12 @@
 SCHEMA = CONFIG["schema"]
 
 
-def refresh_termhub_core_cset_derived_tables_exec(con: Connection, schema: str):
+def refresh_derived_tables(con: Connection, ddl_modules=CORE_CSET_DDL_MODULES, schema=SCHEMA):
```

Comment:
### General purpose `refresh_derived_tables()`
Previously `refresh_termhub_core_cset_derived_tables()`, this new function is general purpose and can refresh any set of derived tables, so long as you pass it `ddl_modules`.

`ddl_modules` was a static list we had to keep up-to-date in this function. Eventually I want to have it auto-computed based on the new `RECURSIVE_DEPENDENT_TABLE_MAP`. But for now I've renamed it to `CORE_CSET_DDL_MODULES` and moved it into `backend/db/config.py`.

Reply:
### `refresh_any_dependent_tables()`
This is a layer on top of `refresh_derived_tables()`. This 
This is now called more directly rather than calling refresh_derived_tables(). Instead of passing the list of dervied tables you want refreshed, you pass the list of tables updated, and it figures out the derived tables to update in the correct order. Also added further utility functions to aid with this: get_dependent_tables_queue() and extract_keys_from_nested_dict().